velox/functions/prestosql/aggregates/SimpleNumericAggregate.h

```diff
@@ -127,11 +128,22 @@ class SimpleNumericAggregate : public exec::Aggregate {
             groups[i], TData(decoded.valueAt<TValue>(i)), updateSingleValue);
       });
     } else if (decoded.isIdentityMapping() && !std::is_same_v<TValue, bool>) {
-      auto data = decoded.data<TValue>();
-      rows.applyToSelected([&](vector_size_t i) {
-        updateNonNullValue<tableHasNulls, TData>(
-            groups[i], TData(data[i]), updateSingleValue);
-      });
+      if constexpr (std::is_same_v<UnscaledLongDecimal, TData>) {
```

Comment:
```c++
      auto data = decoded.data<char>();
      rows.applyToSelected([&](vector_size_t i) {
        updateNonNullValue<tableHasNulls, TData>(
            groups[i], TData(deserialize<TValue>(data + i * sizeof(TValue))), updateSingleValue);
      });
```

Reply:
@karteekmurthys Have you tried this?  This way we can centralize the `if` check inside `deserialize`

Reply:
@Yuhta I tried this. `decoded.data<char>();` returns const char* which cannot be used in the deserialize<TValue>() method. We manipulate the char * to get to the current offset.

Reply:
We should accept `const char*` in `deserialize`.  Add `const T* Aggregate::value(const char* group)` if you need.

Reply:
There too many references to value<> and this must be a separate effort.

```
/Users/karteek/workspace/velox/velox/functions/prestosql/aggregates/ArrayAggAggregate.cpp:128:7: error: 'this' argument to member function 'appendValue' has type 'const facebook::velox::aggregate::ValueList', but function is not marked const
      values.appendValue(decodedElements_, row, allocator_);
      ^~~~~~
/Users/karteek/workspace/velox/./velox/functions/prestosql/aggregates/ValueList.h:31:8: note: 'appendValue' declared here
  void appendValue(
       ^
/Users/karteek/workspace/velox/velox/functions/prestosql/aggregates/ArrayAggAggregate.cpp:144:7: error: 'this' argument to member function 'appendRange' has type 'const facebook::velox::aggregate::ValueList', but function is not marked const
      values.appendRange(
      ^~~~~~
/Users/karteek/workspace/velox/./velox/functions/prestosql/aggregates/ValueList.h:36:8: note: 'appendRange' declared here
  void appendRange(
       ^
/Users/karteek/workspace/velox/velox/functions/prestosql/aggregates/ArrayAggAggregate.cpp:154:7: error: 'this' argument to member function 'free' has type 'const facebook::velox::aggregate::ValueList', but function is not marked const
      value<ArrayAccumulator>(group)->elements.free(allocator_);

```

Reply:
You can have both `T* value(char* group)` and `const T* value(const char* group)`

Reply:
Done.